scripts/upload_ci_stats.sh

```diff
@@ -0,0 +1,178 @@
+    commit_sha text,
+    job_name text,
+    repository text,
+    ref_name text,
+    event_name text,
+    pr_number int,
+    job_status text,
+    url text,
+    run_attempt int,
+    run_id bigint,
+    run_number int
+);
+
+create unique index on job(job_date);
+
+select create_hypertable('job', 'job_date');
+
+create table if not exists test(
+    job_date timestamptz,
+    test_name text,
+    test_status text,
+    test_duration float
+);
+
+create unique index on test(job_date, test_name);
+
+select create_hypertable('test', 'job_date');
+
+create table log(
+    job_date timestamptz,
+    test_name text,
+    contents text
+);
+
+create unique index on log(job_date, log_name);
+
+select create_hypertable('log', 'job_date');
+
+-- don't add a trailing newline because bash command substitution removes it"
+
+DROP_QUERY="
+drop table test;
+drop table job;
+drop table log;
+"
+
+# Recreate the tables if the schema changed.
+EXISTING_SCHEMA=$("${PSQL[@]}" -qtAXc "
+    create table if not exists _schema(create_query text, drop_query text);
+    select create_query from _schema;
+")
+
+if ! [ "${EXISTING_SCHEMA}" == "${DESIRED_SCHEMA}" ];
+then
+    "${PSQL[@]}" -v create="$DESIRED_SCHEMA" -v drop="$DROP_QUERY" <<EOF
+\set ON_ERROR_STOP 0
+select drop_query from _schema
+\gexec
+:drop
+:create
+truncate table _schema;
+insert into _schema values (:'create', :'drop');
+EOF
+fi
+
+# Create the job record.
+COMMIT_SHA=$(git -C "$(dirname "${BASH_SOURCE[0]}")" rev-parse @)
+export COMMIT_SHA
+
+JOB_NAME="${JOB_NAME:-test-job}"
+export JOB_NAME
+
+JOB_DATE=$("${PSQL[@]}" -qtAX <<<"
+insert into job values (
+    now(), '$COMMIT_SHA', '$JOB_NAME',
+    '$GITHUB_REPOSITORY', '$GITHUB_REF_NAME', '$GITHUB_EVENT_NAME',
+    '$GITHUB_PR_NUMBER', '$JOB_STATUS',
+    'https://github.com/timescale/timescaledb/actions/runs/$GITHUB_RUN_ID/attempts/$GITHUB_RUN_ATTEMPT',
+    '$GITHUB_RUN_ATTEMPT', '$GITHUB_RUN_ID', '$GITHUB_RUN_NUMBER')
+returning job_date;
+")
+export JOB_DATE
+
+# Split the regression.diffs into per-test files.
+gawk '
+    match($0, /^(diff|\+\+\+|\-\-\-) .*\/(.*)[.]out/, a) {
+        file = a[2] ".diff";
+        next;
+    }
+
+    { if (file) print $0 > file; }
+' regression.log
+
+# Snip the long sequences of "+" or "-" changes in the diffs.
+for x in *.diff;
+do
+    if ! [ -e "$x" ] ; then continue ; fi
+    gawk -v max_context_lines=10 -v min_context_lines=2 '
+        /^-/ { new_sign = "-" }
+        /^+/ { new_sign = "+" }
+        /^[^+-]/ { new_sign = " " }
+
+        {
+            if (old_sign != new_sign) {
+                to_print = lines_buffered > max_context_lines ? min_context_lines : lines_buffered;
+
+                if (lines_buffered > to_print)
+                    print "<" lines_buffered - to_print " lines skipped>";
+
+                for (i = 0; i < to_print; i++) {
+                    print buf[(NR + i - to_print) % max_context_lines]
+                }
+
+                printf("c %04d: %s\n", NR, $0);
```

Comment:
The `c %04d: ` part is for debugging. I'd like to leave it in for a while, until I'm absolutely sure that this diff snipping code works. It doesn't harm readability too much, and actually helps notice errors, because it has line numbers.